src/sentry/api/endpoints/project_stacktrace_coverage.py

```diff
@@ -0,0 +1,79 @@
+from sentry.api.bases.project import ProjectEndpoint
+from sentry.api.endpoints.project_stacktrace_link import generate_context, get_stacktrace_config
+from sentry.api.serializers import serialize
+from sentry.integrations.utils.code_mapping import get_sorted_code_mapping_configs
+from sentry.integrations.utils.codecov import codecov_enabled, fetch_codecov_data
+from sentry.models.project import Project
+
+
+@region_silo_endpoint
+class ProjectStacktraceCoverageEndpoint(ProjectEndpoint):
+    publish_status = {
+        "GET": ApiPublishStatus.PRIVATE,
+    }
+    """
+    Returns codecov data for a given stacktrace.
+    Similar to stacktrace-link, but for coverage data.
+
+    `file`: The file path from the stack trace
+    `commitId` (optional): The commit_id for the last commit of the
+                           release associated to the stack trace's event
+    `sdkName` (optional): The sdk.name associated with the event
+    `absPath` (optional): The abs_path field value of the relevant stack frame
+    `module`  (optional): The module field value of the relevant stack frame
+    `package` (optional): The package field value of the relevant stack frame
+    `groupId` (optional): The Issue's id.
+    """
+
+    owner = ApiOwner.ISSUES
+
+    def get(self, request: Request, project: Project) -> Response:
+        should_get_coverage = codecov_enabled(project.organization)
+        if not should_get_coverage:
+            return Response({"detail": "Codecov not enabled"}, status=400)
+
+        ctx = generate_context(request.GET)
+        filepath = ctx.get("file")
+        if not filepath:
+            return Response({"detail": "Filepath is required"}, status=400)
+
+        configs = get_sorted_code_mapping_configs(project)
+        if not configs:
+            return Response({"detail": "No code mappings found for this project"}, status=400)
+
+        error = None
+        serialized_config = None
+
+        with configure_scope() as scope:
+            result = get_stacktrace_config(configs, ctx)
+            error = result["error"]
+
+            # Post-processing before exiting scope context
+            if result["current_config"]:
+                serialized_config = serialize(result["current_config"]["config"], request.user)
+                provider = serialized_config["provider"]["key"]
+                # Use the provider key to split up stacktrace-link metrics by integration type
+                scope.set_tag("integration_provider", provider)  # e.g. github
+                codecov_data = fetch_codecov_data(
+                    config={
+                        "repository": result["current_config"]["repository"],
+                        "config": serialized_config,
+                        "outcome": result["current_config"]["outcome"],
+                    }
+                )
+                return Response(codecov_data)
+
+        return Response(
+            {
+                "error": error,
+            }
+        )
```

Comment:
Should return a status code here too, maybe 500?

Reply:
returning 400 because it could be a few things